client/src/app/components/page/register/register.component.html

```diff
@@ -0,0 +1,26 @@
+<div class="formContainer">
+  <form [formGroup]="registerForm" (submit)="onSubmit()">
+    <label>
+      UserName<br>
+      <input type="text" formControlName="userName">
+      <div *ngIf="registerForm.controls['userName'].invalid && (registerForm.controls['userName'].dirty || registerForm.controls['userName'].touched)" 
```

Comment:
please put all this conditions to method hasErrors(controleName: string): boolean {...}
and reuse it